src/test/java/plugins/SshSlavesPluginTest.java

```diff
@@ -58,35 +55,74 @@
     }
 
     @Test public void connectWithPassword() {
-        SshSlaveLauncher launcher = slave.setLauncher(SshSlaveLauncher.class);
-        launcher.host.set(sshd.ipBound(22));
-        launcher.port(sshd.port(22));
-
-        SshCredentialDialog dia = launcher.addCredential();
-        UserPwdCredential cred = dia.select(UserPwdCredential.class);
-        cred.username.set("test");
-        cred.password.set("test");
-        cred.add();
+        configureDefaultSSHSlaveLauncher().pwdCredentials("test", "test");
         slave.save();
 
         verify();
     }
 
     @Test public void connectWithKey() {
-        SshSlaveLauncher launcher = slave.setLauncher(SshSlaveLauncher.class);
-        launcher.host.set(sshd.ipBound(22));
-        launcher.port(sshd.port(22));
-
-        SshCredentialDialog dia = launcher.addCredential();
-        SshPrivateKeyCredential cred = dia.select(SshPrivateKeyCredential.class);
-        cred.username.set("test");
-        cred.enterDirectly(sshd.getPrivateKeyString());
-        cred.add();
+        configureDefaultSSHSlaveLauncher().keyCredentials("test", sshd.getPrivateKeyString());
+        // To avoid test flakiness
+        elasticSleep(1000);
```

Comment:
@jtnord Without waiting some time to save the configuration this is what I get in the slave configuration xml:

```
...
  <launcher class="hudson.plugins.sshslaves.SSHLauncher" plugin="ssh-slaves@1.10">
    <host>192.168.99.100</host>
    <port>32859</port>
    <maxNumRetries>0</maxNumRetries>
    <retryWaitTime>0</retryWaitTime>
  </launcher>
...
```

If I then re-save the configuration, or if I wait for some time before actually saving the slave configuration, I get this instead:

```
...
  <launcher class="hudson.plugins.sshslaves.SSHLauncher" plugin="ssh-slaves@1.10">
    <host>192.168.99.100</host>
    <port>32859</port>
    <credentialsId>c2605d23-3e92-4c64-907d-5d90f2cdfd0f</credentialsId>
    <maxNumRetries>0</maxNumRetries>
    <retryWaitTime>0</retryWaitTime>
  </launcher>
...
```

And that's happening because the DOM is not generated quick enough and a `null` is sent for the credentials field. I don't see an actual issue here, just a test that needs to take this into account and has to wait until the dropdown has been filled with the credential that has just been created. So that is what I have done instead of sleeping for 1 second which, I agree, is not very elegant nor reliable. If you see it differently, please do let me know.

Other comments are also addressed.